dbsnp-importer/src/main/java/uk/ac/ebi/eva/dbsnpimporter/jobs/steps/processors/AssemblyCheckFilterProcessor.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2017 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.ac.ebi.eva.dbsnpimporter.jobs.steps.processors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.batch.item.ItemProcessor;
+
+import uk.ac.ebi.eva.commons.core.models.Region;
+import uk.ac.ebi.eva.dbsnpimporter.models.SubSnpCoreFields;
+import uk.ac.ebi.eva.dbsnpimporter.sequence.SequenceReader;
+
+import java.util.NoSuchElementException;
+
+public class AssemblyCheckFilterProcessor implements ItemProcessor<SubSnpCoreFields, SubSnpCoreFields> {
+
+    private static final Logger logger = LoggerFactory.getLogger(AssemblyCheckFilterProcessor.class);
+
+    private SequenceReader assemblyReader;
+
+    public AssemblyCheckFilterProcessor(SequenceReader assemblyReader) {
+        this.assemblyReader = assemblyReader;
+    }
+
+    @Override
+    public SubSnpCoreFields process(SubSnpCoreFields subSnpCoreFields) throws Exception {
+        String referenceAllele = subSnpCoreFields.getReferenceInForwardStrand();
+        // check if reference is not empty
+        if (referenceIsNotEmpty(referenceAllele)) {
```

Comment:
What about changing this to positive? It is usually easier to understand. I would also take out the "reference" from the method name because it could be used for any string.

```java
if (isEmpty(referenceAllele)) {
    return subSnpCoreFields;
}

Region region = ...
```

Reply:
👍